tests/sentry/incidents/endpoints/test_organization_alert_rule_index.py

```diff
@@ -1640,6 +1641,59 @@ def test_performance_alert(self, record_analytics: MagicMock) -> None:
             resp = self.get_response(self.organization.slug, **valid_alert_rule)
             assert resp.status_code == 201
 
+    @with_feature("organizations:incidents")
+    @with_feature("organizations:workflow-engine-metric-detector-limit")
+    @patch("sentry.quotas.backend.get_metric_detector_limit")
+    def test_metric_alert_limit(self, mock_get_limit: MagicMock) -> None:
+        # Set limit to 2 alert rules
+        mock_get_limit.return_value = 2
+
+        # Create 2 existing metric alert rules (1 active, 1 to be deleted)
+        self.create_alert_rule(organization=self.organization, status=AlertRuleStatus.PENDING)
+        self.create_alert_rule(organization=self.organization, status=AlertRuleStatus.SNAPSHOT)
```

Comment:
nit: You could just do 

```
alert_rule = self.create_alert_rule(organization=self.organization)
alert_rule.status = AlertRuleStatus.SNAPSHOT.value
.alert_rule.save()
```

instead of updating `create_alert_rule` and the test util for one test